### ISSUE TITLE
Cling symbols not exported in win32 and win64

### ISSUE THREAD

**Issue by shangjiaxuan** [root-project/root#10452]
- [x] Checked for duplicates

### Describe the bug
Cling symbol `cling::runtime::internal::LifetimeHandler` may need to be explicitly exported from shared library on windows.

Define a function with function call (`sqrt` in my scenario) in interactive mode.
Use it to initialize a `TF1`
nightly 6.27.01 outputs following:
(6.26.00 release crash immediately)
```
IncrementalExecutor::executeFunction: symbol '??0LifetimeHandler@internal@runtime@cling@@QAE@PAVDynamicExprInfo@123@PAVDeclContext@clang@@PBDPAVInterpreter@3@@Z' unresolved while linking function '_GLOBAL__sub_I_cling_module_16'!
You are probably missing the definition of public: __thiscall cling::runtime::internal::LifetimeHandler::LifetimeHandler(class cling::runtime::internal::DynamicExprInfo *,class clang::DeclContext *,char const *,class cling::Interpreter *)
Maybe you need to load the corresponding shared library?
IncrementalExecutor::executeFunction: symbol '??1LifetimeHandler@internal@runtime@cling@@QAE@XZ' unresolved while linking function '_GLOBAL__sub_I_cling_module_16'!
You are probably missing the definition of public: __thiscall cling::runtime::internal::LifetimeHandler::~LifetimeHandler(void)
Maybe you need to load the corresponding shared library?
```
Then call `TF1::Draw()` on it (if 6.27.01), and see interpreter dump stacktrace.

Problem not present in script mode. Compiling with `.L test.cpp+` fails, but that seems irrelevant.

Without `sqrt` math function interpreter works (function returing 0 immediately).
### Expected behavior
`TF1` initialized with function pointer.

### To Reproduce
Detailed in description section

### Setup
root nightly 6.27.01 32 bit and 64 bit
root release 6.26.00 32 bit
windows 64 bit os
installed from [nightly repo](https://root.cern/download/nightly/) and the release page binary.

### Additional context
Should be enough.

**Comment by bellenot** [comment]
Can you provide a concrete example showing the problem? (just copy and paste your session here)

**Comment by shangjiaxuan** [comment]
Example code:
```
constexpr size_t num = 1000;
constexpr size_t num_bin = 20;
double fit_fcn(double* _x, double* _p)
{
	double x = *_x;
	double a = *_p;
	return double(num) / num_bin * (a + 1) * (a + 1) / ((2 + a * 2.0 / 3) * sqrt((1 + x) * (1 - a * x) * (1 - a * x) * (1 - a * x) * (1 - a * x) * (1 - a * x)));
}
TF1 f1("f1", fit_fcn, -1, 0, 1);
f1.Draw()
```
Also noted that creating function with `TF1* f1 = new TF1("f1", fit_fcn, -1, 0, 1);` will result in `input_line_13:2:18: error: use of undeclared identifier 'fit_fcn'`

**Comment by bellenot** [comment]
Thanks @shangjiaxuan ! This is now fixed in the master branch by the commit #1e21896
